test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -494,36 +497,54 @@ static int open_pktios(void)
 		}
 
 		pools[iface] = pkt_pool;
-		pktio = odp_pktio_open(iface_name[iface], pkt_pool,
-				       &pktio_param);
-		if (pktio == ODP_PKTIO_INVALID)
-			pktio = odp_pktio_lookup(iface_name[iface]);
+
+		/* Zero'th device is always PKTOUT TM as we use it from XMIT */
+		if (iface == 0) {
+			pktio_param.out_mode = ODP_PKTOUT_MODE_TM;
+
+			pktio = odp_pktio_open(iface_name[iface], pkt_pool,
+					       &pktio_param);
+
+			/* On failure check if pktio can be opened in non-TM mode.
+			 * If non-TM mode works, then we can assume that PKTIO
+			 * does not support TM
+			 */
+			if (pktio == ODP_PKTIO_INVALID) {
+				pktio_param.out_mode = ODP_PKTOUT_MODE_DIRECT;
+				pktio = odp_pktio_open(iface_name[iface], pkt_pool,
+						       &pktio_param);
+
+				/* Return +ve to indicate no TM support */
```

Comment:
It took me a moment to understand what '+ve' stands for. Maybe it could be replaces with '>0'.